gammapy/modeling/parameter.py

```diff
@@ -208,6 +209,7 @@ def value(self):
     @value.setter
     def value(self, val):
         self._factor = float(val) / self._scale
+        self._check_value_within_bounds()
```

Comment:
During fitting we actually go through the `factor` setter, so probably the warning is never emitted. To avoid the code duplication I would propose to change move the check to the `factor` setter and change the `value` setter to call `self.factor = float(val) / self._scale`...

Reply:
OK, I didn't know that. However, it seems that the warning is already emitted (check cell [22] of https://github.com/luca-giunti/share/blob/master/analysis_1.ipynb) from the value setter somehow. But if you prefer, I will do as you propose

Reply:
Probably the warning is emitted on I/O already, but I think not during fitting...